zilliqa/src/block_store.rs

```diff
@@ -132,8 +132,10 @@ impl BlockCache {
     /// (note that this will be BLOCK_CACHE_HEAD_BUFFER_ENTRIES >> shift cached views, since the
     /// head cache is set associative)
     pub fn min_head_cache_key(&self, highest_known_view: u64) -> u128 {
+        let delta = u128::try_from(constants::BLOCK_CACHE_HEAD_BUFFER_ENTRIES).unwrap();
         let highest_key = u128::from(highest_known_view + 1) << self.shift;
-        highest_key - u128::try_from(constants::BLOCK_CACHE_HEAD_BUFFER_ENTRIES).unwrap()
```

Comment:
This fixes a bunch of failing tests in `cargo test` for me.

Reply:
"oops!" - sorry! Well caught ... I wonder why that didn't panic in dev mode for me?

Reply:
https://github.com/Zilliqa/zq2/pull/1703